addons/website_sale_delivery_mondialrelay/controllers/controllers.py

```diff
@@ -80,7 +80,7 @@ class PaymentPortalMondialRelay(PaymentPortal):
     @http.route()
     def shop_payment_transaction(self, *args, **kwargs):
         order = request.website.sale_get_order()
-        if order.partner_shipping_id.is_mondialrelay and not order.carrier_id.is_mondialrelay:
+        if order.partner_shipping_id.is_mondialrelay and not order.carrier_id.is_mondialrelay and order.delivery_set:
```

Comment:
We should definitely check that `carrier_id` is set as well. It might make the `order.delivery_set` condition superfluous but we can keep it as it clearly shows the intent of this check.
```suggestion
        if order.partner_shipping_id.is_mondialrelay and order.carrier_id and not order.carrier_id.is_mondialrelay and order.delivery_set:
```